### DISCUSSION TITLE
How to reference filter from image in code behind

### DISCUSSION THREAD

**Discussion by QadiymStewart** [vvvv/SVG#847] [Q&A]
How to reference filter in code behind. Trying to apply the filter to the image. Builing the svg from code. My filter has an id of     ID = "a". How to I reference it in the SVGImage  ->   Filter = new Uri(),

     SvgDocument svgdoc = new SvgDocument
            {
                AspectRatio = new SvgAspectRatio
                {
                    Align = SvgPreserveAspectRatio.xMidYMid,
                    Slice = true
                },

                Height = SVGHeight,
                Width = SVGWidth,
            };

            var filter = new SvgFilter
            {
                ID = "a",
                ColorInterpolation = SvgColourInterpolation.SRGB
            };

            var colorMatrix = new SvgColourMatrix
            {
                Result = "",
                Values = ".33 .33 .33 0 0 .33 .33 .33 0 0 .33 .33 .33 0 0 0 0 0 1 0",
                Input = "SourceGraphic"
            };

            var feComponentTransfer = new SvgComponentTransfer
            {
                Result = "duotone_canvas"
            };

            SvgFuncR feFuncR = new SvgFuncR
            {
                Type = SvgComponentTransferType.Table,
                TableValues = ConvertStringToTable(RTable)
            };
            SvgFuncG feFuncG = new SvgFuncG
            {
                Type = SvgComponentTransferType.Table,
                TableValues = ConvertStringToTable(GTable)
            };
            SvgFuncB feFuncB = new SvgFuncB
            {
                Type = SvgComponentTransferType.Table,
                TableValues = ConvertStringToTable(BTable)
            };
            SvgFuncA feFuncA = new SvgFuncA
            {
                Type = SvgComponentTransferType.Table,
                TableValues = ConvertStringToTable(ATable)
            };

            feComponentTransfer.Children.Add(feFuncR);
            feComponentTransfer.Children.Add(feFuncG);
            feComponentTransfer.Children.Add(feFuncB);
            feComponentTransfer.Children.Add(feFuncA);

            // var uriBuilder = new UriBuilder("#a");

            var svgImage = new SvgImage
            {
                Height = SVGHeight,
                Width = SVGWidth,
                AspectRatio = new SvgAspectRatio
                {
                    Align = SvgPreserveAspectRatio.xMidYMid,
                    Slice = true
                },
                Filter = new Uri(),
                Href = DataUrl
            };

            filter.Children.Add(colorMatrix);
            filter.Children.Add(feComponentTransfer);
            svgdoc.Children.Add(filter);
            svgdoc.Children.Add(svgImage);

            var svg = new SKSvg();
            var newDoc = svg.FromSvgDocument(svgdoc);

**Comment by wieslawsoltes** [comment]
I use this helper function:
```C#
        public static T? GetReference<T>(this SvgElement svgElement, Uri? uri) where T : SvgElement
        {
            if (uri is null)
            {
                return default;
            }

            var svgElementById = svgElement.OwnerDocument?.GetElementById(uri.ToString());
            if (svgElementById is { })
            {
                return svgElementById as T;
            }

            return default;
        }
```
